### PR TITLE
Move QueryNotAdequatelyPushedDownErrorCode to PinotErrorCode

### PR DESCRIPTION
**Pull request by fx19880617**
Per slack discussion, Pinot connector accidentally took the error code space `0x0625`. 

This PR moves the corresponding error to PinotErrorCode space `0x0505`.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by fx19880617** [comment]
@wenleix @agrawaldevesh

File: presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotErrorCode.java [hunk 1]
```diff
@@ -36,6 +37,7 @@
     PINOT_INVALID_CONFIGURATION(10, INTERNAL_ERROR),
     PINOT_DATA_FETCH_EXCEPTION(11, EXTERNAL, true),
     PINOT_REQUEST_GENERATOR_FAILURE(12, INTERNAL_ERROR), // Unable to generate Pinot request
+    PINOT_PUSH_DOWN_QUERY_NOT_PRESENT(20, USER_ERROR),
```
Comment (agrawaldevesh) [inline presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotErrorCode.java:40]: Good idea to not use 13 :-P

File: presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotErrorCode.java [hunk 2]
```diff
@@ -36,6 +37,7 @@
     PINOT_INVALID_CONFIGURATION(10, INTERNAL_ERROR),
     PINOT_DATA_FETCH_EXCEPTION(11, EXTERNAL, true),
     PINOT_REQUEST_GENERATOR_FAILURE(12, INTERNAL_ERROR), // Unable to generate Pinot request
```
Comment (agrawaldevesh) [inline presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotErrorCode.java:39]: Can you also delete the comment for this line. It is redundant.

File: presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotSplitManager.java [hunk 3]
```diff
@@ -183,7 +160,7 @@ public ConnectorSplitSource getSplits(
     {
         PinotTableLayoutHandle pinotLayoutHandle = (PinotTableLayoutHandle) layout;
         PinotTableHandle pinotTableHandle = pinotLayoutHandle.getTable();
-        Supplier<PrestoException> errorSupplier = () -> new QueryNotAdequatelyPushedDownException(QueryNotAdequatelyPushedDownErrorCode.PQL_NOT_PRESENT, pinotTableHandle, connectorId);
+        Supplier<PrestoException> errorSupplier = () -> new QueryNotAdequatelyPushedDownException(PinotErrorCode.PINOT_PUSH_DOWN_QUERY_NOT_PRESENT, pinotTableHandle, connectorId);
```
Comment (agrawaldevesh) [inline presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotSplitManager.java:163]: Is `QueryNotAdequatelyPushedDownException` created any where else ? Should we bury the error code `PinotErrorCode.PINOT_PUSH_DOWN_QUERY_NOT_PRESENT` inside the constructor then ?

File: presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotErrorCode.java (hunk 1 above)
Comment (fx19880617) [reply presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotErrorCode.java:40]: :) wanna start the pushdown related error code with 20+

File: presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotSplitManager.java (hunk 3 above)
Comment (fx19880617) [reply presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotSplitManager.java:163]: My feeling  is that this error code is somehow very high level. We should have more detailed errors in the future. So prefer having an error code here.

File: presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotErrorCode.java (hunk 2 above)
Comment (fx19880617) [reply presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotErrorCode.java]: removed

**Comment by wenleix** [comment]
merged #15022, thanks for the contribution!

**Pull request merged**